<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>T3_RNAseq • franciscomartinez</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="T3_RNAseq">
<meta property="og:description" content="franciscomartinez">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">franciscomartinez</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/T1_Microarray.html">T1_Microarray</a>
    </li>
    <li>
      <a href="../articles/T2_ExprDiferencial.html">T2_ExprDiferencial</a>
    </li>
    <li>
      <a href="../articles/T3_RNAseq.html">T3_RNAseq</a>
    </li>
    <li>
      <a href="../articles/T4_Sobrerrepresentacion.html">T4_Sobrerrepresentacion</a>
    </li>
    <li>
      <a href="../articles/T5_GeneSetAnalysis.html">T5_GeneSetAnalysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="T3_RNAseq_files/header-attrs-2.1/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>T3_RNAseq</h1>
                        <h4 class="author">Francisco Martínez Picó</h4>
            
            <h4 class="date">5/23/2020</h4>
      
      
      <div class="hidden name"><code>T3_RNAseq.Rmd</code></div>

    </div>

    
    
<div id="explicación-del-experimento" class="section level2">
<h2 class="hasAnchor">
<a href="#explicaci%C3%B3n-del-experimento" class="anchor"></a>Explicación del experimento</h2>
<p>Estos datos provienen de un estudio realizado sobre el nematodo <em>Caenorhabditis elegans</em>. Este organismo mide aproximadamente 1 mm de longitud y vive en ambientes templados. <em>C. elegans</em> es utilizado ampliamente como modelo experimental en Biología. Además, fue el primer organismo multicelular cuyo genoma pudo secuenciarse al completo. Esto sucedió en 1998.</p>
<p>Más concretamente, en este estudio se trata de <strong>analizar el efecto de la dieta de <em>C. elegans</em> sobre su metabolismo y crecimiento</strong>. Por tanto, se compara la expresión génica en <strong>tres diferentes grupos</strong> establecidos, contando cada uno de ellos con tres réplicas. Los tres grupos definidos consisten en <em>C. elegans</em> crecido en un cultivo con:</p>
<p>1: <strong>E. coli OP50</strong> 2: <strong>Chryseobacterium sp. CHNTR56</strong> 3: <strong>Comamonas sp. 12022</strong></p>
<p>La relevancia de este estudio subyace en que aporta más información sobre <em>C. elegans</em> como organismo modelo. Sin embargo, aún no se ha publicado ningún artículo con estos datos. Los datos fueron depositados en el NCBI por Orcun Hacariz (<em>Institute of Parasitology</em>, <em>McGill University</em>) el 16 de enero de 2020.</p>
<p>Los datos utilizados en esta viñeta se encuentran en <strong>BioProject código PRJNA601724</strong> y en <strong>GEO código GSE143794</strong>. Los links son los siguientes:</p>
<p>BioProject: <a href="https://www.ncbi.nlm.nih.gov/bioproject/601724" class="uri">https://www.ncbi.nlm.nih.gov/bioproject/601724</a></p>
<p>GEO: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE143794" class="uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE143794</a></p>
<p>Se pueden obtener las runs del mismo en el siguiente link:</p>
<p>SRA run selector: <a href="https://www.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA601724&amp;o=acc_s%3Aa" class="uri">https://www.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA601724&amp;o=acc_s%3Aa</a></p>
</div>
<div id="cargando-paquetes" class="section level2">
<h2 class="hasAnchor">
<a href="#cargando-paquetes" class="anchor"></a>Cargando paquetes</h2>
<p>Antes de empezar, cargamos los paquetes necesarios:</p>
<ul>
<li>
<em>Biobase</em>: este paquete para trabajar con Bioconductor contiene estructuras estandarizadas de datos para representar información genómica.</li>
<li>
<em>SummarizedExperiment</em>: paquete utilizado para trabajar con el objeto RangedSummarizedExperiment, que contendrá toda la información de nuestro experimento de RNAseq.</li>
<li>
<em>Rsamtools</em>: utilizado para obtener la matriz de conteos (objeto RangedSummarizedExperiment) a partir de los archivos .bam.</li>
<li>
<em>GenomicFeatures</em>: contiene una serie de herramientas y métodos para construir y manicular anotaciones de tránscritos.</li>
<li>
<em>GenomicAlignments</em>: utilizado para manipular los alineamientos de RNAseq.</li>
<li>
<em>edgeR</em>: lo utilizaremos para llevar a cabo el análisis de la expresión diferencial (e, implícitamente, llevar a cabo la normalización de los datos).</li>
<li>
<em>ReportingTools</em>: se utiliza para crear el informe con los genes significativos.</li>
<li>
<em>franciscomartinez</em>: para utilizar las funciones que crean las URL a partir de los ID.</li>
<li>
<em>org.Ce.eg.db</em>: para la anotación de los genes y hallar la correspondencia con el id de entrez</li>
<li>
<em>AnnotationDbi</em>: para hacer la anotación de los genes.</li>
</ul>
</div>
<div id="dataset-obtención-de-los-datos" class="section level2">
<h2 class="hasAnchor">
<a href="#dataset-obtenci%C3%B3n-de-los-datos" class="anchor"></a>Dataset (obtención de los datos)</h2>
<p>Todo el proceso de obteción y procesado de datos se realizó sobre el ordenador del master, <strong>masterbio</strong>.</p>
<p>Los archivos .sra que contienen las runs utilizadas son las siguientes:</p>
<ul>
<li>SRR10902994 —&gt; <em>C. elegans</em> crecido en un cultivo con <strong>E. coli OP50</strong>
</li>
<li>SRR10902995 —&gt; <em>C. elegans</em> crecido en un cultivo con <strong>E. coli OP50</strong>
</li>
<li>SRR10902996 —&gt; <em>C. elegans</em> crecido en un cultivo con <strong>E. coli OP50</strong>
</li>
<li>SRR10902997 —&gt; <em>C. elegans</em> crecido en un cultivo con <strong>Chryseobacterium sp. CHNTR56</strong>
</li>
<li>SRR10902998 —&gt; <em>C. elegans</em> crecido en un cultivo con <strong>Chryseobacterium sp. CHNTR56</strong>
</li>
<li>SRR10902999 —&gt; <em>C. elegans</em> crecido en un cultivo con <strong>Chryseobacterium sp. CHNTR56</strong>
</li>
<li>SRR10903000 —&gt; <em>C. elegans</em> crecido en un cultivo con <strong>Comamonas sp. 12022</strong>
</li>
<li>SRR10903001 —&gt; <em>C. elegans</em> crecido en un cultivo con <strong>Comamonas sp. 12022</strong>
</li>
<li>SRR10903002 —&gt; <em>C. elegans</em> crecido en un cultivo con <strong>Comamonas sp. 12022</strong>
</li>
</ul>
<p>En primer lugar, se descargaron los 9 archivos .sra con el comando <em>wget</em>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">wget</span> https://sra-download.ncbi.nlm.nih.gov/traces/sra58/SRR/010647/SRR10902994</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">wget</span> https://sra-download.ncbi.nlm.nih.gov/traces/sra57/SRR/010647/SRR10902995</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">wget</span> https://sra-download.ncbi.nlm.nih.gov/traces/sra55/SRR/010647/SRR10902996</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">wget</span> https://sra-download.ncbi.nlm.nih.gov/traces/sra56/SRR/010647/SRR10902997</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">wget</span> https://sra-download.ncbi.nlm.nih.gov/traces/sra39/SRR/010647/SRR10902998</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">wget</span> https://sra-download.ncbi.nlm.nih.gov/traces/sra36/SRR/010647/SRR10902999</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">wget</span> https://sra-download.ncbi.nlm.nih.gov/traces/sra55/SRR/010647/SRR10903000</span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="fu">wget</span> https://sra-download.ncbi.nlm.nih.gov/traces/sra36/SRR/010647/SRR10903001</span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="fu">wget</span> https://sra-download.ncbi.nlm.nih.gov/traces/sra19/SRR/010647/SRR10903002</span></code></pre></div>
<p>Seguidamente, se obtuvieron los archivos FASTQ utilizando el comando <em>fastq-dump</em> (nuestras runs .sra cuentan con 1 read por archivo).</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1"></a><span class="ex">fastq-dump</span> SRR10902994.sra</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="ex">fastq-dump</span> SRR10902995.sra</span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="ex">fastq-dump</span> SRR10902996.sra</span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="ex">fastq-dump</span> SRR10902997.sra</span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="ex">fastq-dump</span> SRR10902998.sra</span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="ex">fastq-dump</span> SRR10902999.sra</span>
<span id="cb2-7"><a href="#cb2-7"></a><span class="ex">fastq-dump</span> SRR10903000.sra</span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="ex">fastq-dump</span> SRR10903001.sra</span>
<span id="cb2-9"><a href="#cb2-9"></a><span class="ex">fastq-dump</span> SRR10903002.sra</span></code></pre></div>
<p>Después se descargó el genoma de referencia de <em>C. elegans</em>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1"></a><span class="fu">wget</span> http://igenomes.illumina.com.s3-website-us-east-1.amazonaws.com/Caenorhabditis_elegans/Ensembl/WBcel235/Caenorhabditis_elegans_Ensembl_WBcel235.tar.gz</span>
<span id="cb3-2"><a href="#cb3-2"></a></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="fu">gzip</span> -d Caenorhabditis_elegans_Ensembl_WBcel235.tar.gz</span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="fu">tar</span> xvf SCaenorhabditis_elegans_Ensembl_WBcel235.tar</span></code></pre></div>
<p>Se realizaron los alineamientos de los FASTQ sobre el mismo utilizando <em>bowtie2 -x -U archivo.fastq -S archivo.sam</em>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1"></a><span class="ex">bowtie2</span> -x /Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/genome -U SRR10902994.fastq -S SRR10902994.sam</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="ex">bowtie2</span> -x /Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/genome -U SRR10902995.fastq -S SRR10902995.sam</span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="ex">bowtie2</span> -x /Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/genome -U SRR10902996.fastq -S SRR10902996.sam</span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="ex">bowtie2</span> -x /Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/genome -U SRR10902997.fastq -S SRR10902997.sam</span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="ex">bowtie2</span> -x /Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/genome -U SRR10902998.fastq -S SRR10902998.sam</span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="ex">bowtie2</span> -x /Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/genome -U SRR10902999.fastq -S SRR10902999.sam</span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="ex">bowtie2</span> -x /Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/genome -U SRR10903000.fastq -S SRR10903000.sam</span>
<span id="cb4-8"><a href="#cb4-8"></a><span class="ex">bowtie2</span> -x /Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/genome -U SRR10903001.fastq -S SRR10903001.sam</span>
<span id="cb4-9"><a href="#cb4-9"></a><span class="ex">bowtie2</span> -x /Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/genome -U SRR10903002.fastq -S SRR10903002.sam</span></code></pre></div>
<p>A continuación, se pasaron estos archivos .sam a .bam con el comando <em>samtools view -bS archivo | samtools sort -o archivo.bam</em>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1"></a><span class="ex">samtools</span> view -bS SRR10902994.sam <span class="kw">|</span> <span class="ex">samtools</span> sort -o SRR10902994</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="ex">samtools</span> view -bS SRR10902995.sam <span class="kw">|</span> <span class="ex">samtools</span> sort -o SRR10902995</span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="ex">samtools</span> view -bS SRR10902996.sam <span class="kw">|</span> <span class="ex">samtools</span> sort -o SRR10902996</span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="ex">samtools</span> view -bS SRR10902997.sam <span class="kw">|</span> <span class="ex">samtools</span> sort -o SRR10902997</span>
<span id="cb5-5"><a href="#cb5-5"></a><span class="ex">samtools</span> view -bS SRR10902998.sam <span class="kw">|</span> <span class="ex">samtools</span> sort -o SRR10902998</span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="ex">samtools</span> view -bS SRR10902999.sam <span class="kw">|</span> <span class="ex">samtools</span> sort -o SRR10902999</span>
<span id="cb5-7"><a href="#cb5-7"></a><span class="ex">samtools</span> view -bS SRR10903000.sam <span class="kw">|</span> <span class="ex">samtools</span> sort -o SRR10903000</span>
<span id="cb5-8"><a href="#cb5-8"></a><span class="ex">samtools</span> view -bS SRR10903001.sam <span class="kw">|</span> <span class="ex">samtools</span> sort -o SRR10903001</span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="ex">samtools</span> view -bS SRR10903002.sam <span class="kw">|</span> <span class="ex">samtools</span> sort -o SRR10903002</span></code></pre></div>
<p>Se creo un archivo con todos los nombres de los archivos .bam utilizando:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1"></a><span class="fu">ls</span> *.bam <span class="op">&gt;</span> BamFiles.txt </span></code></pre></div>
<p>El contenido de BamFiles.txt era el siguiente:</p>
<p>SRR10902994.bam SRR10902995.bam SRR10902996.bam SRR10902997.bam SRR10902998.bam SRR10902999.bam SRR10903000.bam SRR10903001.bam SRR10903002.bam</p>
<p>Y, finalmente, para crear la matriz de conteos, se utilizó Rsamtools. El código utilizado es el siguiente.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">Rsamtools</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">GenomicFeatures</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">GenomicAlignments</span>)
<span class="no">gtfFile</span> <span class="kw">=</span> <span class="st">"/Caenorhabditis_elegans/Ensembl/WBcel235/Annotation/Genes/genes.gtf"</span>
<span class="no">txdb</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/makeTxDbFromGFF.html">makeTxDbFromGFF</a></span>(<span class="no">gtfFile</span>, <span class="kw">format</span><span class="kw">=</span><span class="st">"gtf"</span>)
<span class="no">genes</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcriptsBy.html">exonsBy</a></span>(<span class="no">txdb</span>, <span class="kw">by</span><span class="kw">=</span><span class="st">"gene"</span>)
<span class="no">dirActualData</span> <span class="kw">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span>(),<span class="st">"/"</span>,<span class="kw">sep</span><span class="kw">=</span><span class="st">""</span>)
<span class="no">sampleTableSingle</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span>(<span class="st">"BamFiles.txt"</span>)
<span class="no">fls</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="no">dirActualData</span>,<span class="no">sampleTableSingle</span>[,<span class="fl">1</span>],<span class="kw">sep</span><span class="kw">=</span><span class="st">""</span>)
<span class="no">bamLst</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/BamFile-class.html">BamFileList</a></span>(<span class="no">fls</span>, <span class="kw">index</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">character</a></span>(),<span class="kw">yieldSize</span><span class="kw">=</span><span class="fl">100000</span>,<span class="kw">obeyQname</span><span class="kw">=</span><span class="fl">TRUE</span>)
<span class="no">PRJNA601724</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicAlignments/man/summarizeOverlaps-methods.html">summarizeOverlaps</a></span>(<span class="kw">features</span> <span class="kw">=</span> <span class="no">genes</span>,<span class="kw">read</span><span class="kw">=</span><span class="no">bamLst</span>,
                                <span class="kw">mode</span><span class="kw">=</span><span class="st">"Union"</span>,
                                <span class="kw">singleEnd</span><span class="kw">=</span><span class="fl">TRUE</span>,
                                <span class="kw">ignore.strand</span><span class="kw">=</span><span class="fl">TRUE</span>,
                                <span class="kw">fragments</span><span class="kw">=</span><span class="fl">FALSE</span>)
<span class="no">Run</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"SRR10902994"</span>,<span class="st">"SRR10902995"</span>,<span class="st">"SRR10902996"</span>,<span class="st">"SRR10902997"</span>,<span class="st">"SRR10902998"</span>,<span class="st">"SRR10902999"</span>,
        <span class="st">"SRR10903000"</span>,<span class="st">"SRR10903001"</span>,<span class="st">"SRR10903002"</span>)
<span class="no">Treatment</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">3</span>,<span class="fl">3</span>)
<span class="no">Treatment</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="no">Treatment</span>,<span class="kw">levels</span><span class="kw">=</span><span class="fl">1</span>:<span class="fl">3</span>,<span class="kw">labels</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"E.coli"</span>,<span class="st">"Chryseobacterium"</span>,<span class="st">"Comamonas"</span>))
<span class="fu">colData</span>(<span class="no">PRJNA601724</span>) <span class="kw">=</span> <span class="fu">DataFrame</span>(<span class="no">Treatment</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">PRJNA601724</span>) <span class="kw">=</span> <span class="no">Run</span>
<span class="fu"><a href="https://rdrr.io/r/base/save.html">save</a></span>(<span class="no">PRJNA601724</span>,<span class="kw">file</span><span class="kw">=</span><span class="st">"PRJNA601724.rda"</span>)</pre></body></html></div>
<p>Como resultado final obtenemos un objeto <strong>SummarizedExperiment::RangedSummarizedExperiment</strong> que contendrá los datos y los metadatos con los que trabajaremos.</p>
</div>
<div id="primera-aproximación-a-los-datos" class="section level2">
<h2 class="hasAnchor">
<a href="#primera-aproximaci%C3%B3n-a-los-datos" class="anchor"></a>Primera aproximación a los datos</h2>
<p>Observamos algunas características de nuestros datos. Contamos con 46.748 filas (o genes) y 9 muestras (recordar que eran tres grupos experimentales con 3 réplicas cada uno).</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">'PRJNA601724'</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">'franciscomartinez'</span>)
<span class="no">se</span> <span class="kw">=</span> <span class="no">PRJNA601724</span>
<span class="co"># dim(se)</span>
<span class="co"># assay(se) # Matriz de conteos</span>
<span class="co"># colData(se) # Datos fenotípicos</span>
<span class="co"># rownames(se) # Nombre de los genes</span>
<span class="co"># rowRanges(se) # Información sobre la secuenciación (sobre las reads)</span>
<span class="co"># rowData(se)</span></pre></body></html></div>
</div>
<div id="filtrado-independiente" class="section level1">
<h1 class="hasAnchor">
<a href="#filtrado-independiente" class="anchor"></a>Filtrado independiente</h1>
<p>Para comenzar, hacemos el sumatorio de los conteos por genes (filas). Como era de esperar, <strong>hay genes no tiene ninguna lectura alineada en ninguna de las muestras</strong>. Pero no sólo eso, además, <strong>el primer cuartil es 0</strong>. Esto quiere decir que <strong>al menos la cuarta parte de las lecturas son 0</strong>. Como nuestro objetivo es hacer un análisis de la expresión diferencial, deberemos eliminar estos datos, ya que no nos aportan información de ningún tipo (probablemente, son consecuencia del proceso de alineamiento).</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">x</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="fu">assay</span>(<span class="no">se</span>),<span class="fl">1</span>,<span class="no">sum</span>)
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span>(<span class="no">x</span>)
<span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">x</span> <span class="kw">==</span> <span class="fl">0</span>) <span class="co"># 23.200 genes no tienen ninguna lectura.</span>
<span class="no">se0</span> <span class="kw">=</span> <span class="no">se</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">x</span> <span class="kw">&gt;</span> <span class="fl">0</span>),] <span class="co"># Se quitan aquellas que en el total de las 9 muestras tienen 0 reads</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">se</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">x</span> <span class="kw">&gt;</span> <span class="fl">0</span>),]) <span class="co"># Nos quedamos con 23.548 genes</span>
<span class="no">se0</span></pre></body></html></div>
<p>Nos quedamos con 23.548 genes. Una vez hecho esto, ya podemos comenzar el proceso de normalización.</p>
</div>
<div id="normalización-o-preprocesado" class="section level1">
<h1 class="hasAnchor">
<a href="#normalizaci%C3%B3n-o-preprocesado" class="anchor"></a>Normalización o preprocesado</h1>
<p>Los problemas o factores no biológicos que aportan ruido y que tratamos de corregir con la normalización de los datos son:</p>
<ol style="list-style-type: decimal">
<li>
<em>Profundidad de secuenciación</em>: tendremos algunos de los genes cuyo conteo en algunas de las muestras será 0, mientras que otros genes contarán con un valor de conteo muy grande. Es decir, <strong>tenemos unos conteos con unas variabilidades muy grandes</strong>. Esto produce un problema a la hora de comparar los datos.</li>
</ol>
<p>Comprobamos la profundidad de lectura entre las muestras:</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="no">y</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="fu">assay</span>(<span class="no">se</span>),<span class="fl">2</span>,<span class="no">sum</span>)
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span>(<span class="no">y</span>) <span class="co"># Profundidad de secuenciación es similar entre muestras.</span></pre></body></html></div>
<ol start="2" style="list-style-type: decimal">
<li><p><em>Composición del RNA</em></p></li>
<li><p><em>Contenido GC</em></p></li>
<li><p><em>Longitud del gen</em>: como en el proceso de secuencicación se trocean los mRNAs, si un gen es más grande se producirán más trozos del mismo y por tanto su conteo final será mayor.</p></li>
</ol>
<p>Para llevar a cabo esta normalización, podríamos haber utilizado el procedimiento <strong>TMM</strong>. Sin embargo, los métodos utilizados para el análisis de expresión diferencial <strong>edgeR</strong> y <strong>DEseq2</strong> ya incorporan de forma nativa un proceso de homogeneización o normalización. Por lo tanto, como para el análisis de la expresión diferencial utilizaremos <strong>edgeR</strong>, también utilizaremos su proceso de normalización integrado.</p>
</div>
<div id="análisis-de-la-expresión-diferencial" class="section level1">
<h1 class="hasAnchor">
<a href="#an%C3%A1lisis-de-la-expresi%C3%B3n-diferencial" class="anchor"></a>Análisis de la expresión diferencial</h1>
<p>Para llevar a cabo el <strong>análisis de la expresión diferencial</strong> utilizaremos el método propuesto en la siguiente bibliografía (el modelo probabilístico que subyace es una binomial negativa):</p>
<ol style="list-style-type: decimal">
<li><p>Robinson, M. D. &amp; Smyth, G. K. Small-sample estimation of negative binomial dispersion, with applications to SAGE data Biostatistics, 2008, 9, 321-332</p></li>
<li><p>Robinson, M. D. &amp; Smyth, G. K. Moderated statistical tests for assessing differences in tag abundance Bioinformatics, 2007, 23, 2881-2887</p></li>
</ol>
<p>Para ello, utilizarenos el <strong>paquete edgeR</strong>. Como tenemos 3 condiciones, haremos las comparaciones 2 a 2.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="co"># 1. E.coli - Chryseobacterium:</span>

<span class="no">sel1</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">PRJNA601724</span>)[,<span class="st">"Treatment"</span>] <span class="kw">==</span> <span class="st">"E.coli"</span> <span class="kw">|</span>
  <span class="fu">colData</span>(<span class="no">PRJNA601724</span>)[,<span class="st">"Treatment"</span>] <span class="kw">==</span> <span class="st">"Chryseobacterium"</span>

<span class="no">sel1</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">sel1</span>)

<span class="no">dge1</span> <span class="kw">=</span> <span class="fu">DGEList</span>(<span class="kw">counts</span> <span class="kw">=</span> <span class="fu">assay</span>(<span class="no">PRJNA601724</span>)[,<span class="no">sel1</span>],
            <span class="kw">group</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">PRJNA601724</span>)[<span class="no">sel1</span>,<span class="st">"Treatment"</span>])

<span class="no">dge1c</span> <span class="kw">=</span>  <span class="fu">estimateCommonDisp</span>(<span class="no">dge1</span>) <span class="co"># Asumimos dispersión común para todos los genes</span>
<span class="no">dge1c</span>$<span class="no">common.dispersion</span> <span class="co"># 0.1519314</span>

<span class="no">dge1t</span> <span class="kw">=</span> <span class="fu">estimateTagwiseDisp</span>(<span class="no">dge1c</span>) <span class="co"># Asumimos una dispersión distinta para cada gen.</span>
<span class="no">dge1t</span>$<span class="no">tagwise.dispersion</span> <span class="co"># Para cada gen</span>

<span class="no">dge1cET</span> <span class="kw">=</span> <span class="fu">exactTest</span>(<span class="no">dge1c</span>)
<span class="no">dge1tET</span> <span class="kw">=</span> <span class="fu">exactTest</span>(<span class="no">dge1t</span>)

<span class="fu">topTags</span>(<span class="no">dge1cET</span>)
<span class="fu">topTags</span>(<span class="no">dge1tET</span>) <span class="co"># Nos centraremos en este.</span>

<span class="no">out1</span> <span class="kw">=</span> <span class="fu">topTags</span>(<span class="no">dge1tET</span>, <span class="kw">n</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="fu">assay</span>(<span class="no">se0</span>)))

<span class="no">p.values1</span> <span class="kw">=</span> <span class="no">out1</span>$<span class="no">table</span>[,<span class="st">"PValue"</span>]
<span class="no">p.values1.BH</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span>(<span class="no">p.values1</span>, <span class="st">"BH"</span>)

<span class="no">wormID</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">out1</span>) <span class="co"># Cogemos los ID de los genes y creamos el link a la URL.</span>

<span class="no">final1</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">GENE</span> <span class="kw">=</span> <span class="no">wormID</span>, <span class="kw">P.VALUE</span> <span class="kw">=</span> <span class="no">out1</span>[,<span class="st">'PValue'</span>], <span class="kw">p.BH</span> <span class="kw">=</span> <span class="no">p.values1.BH</span>)

<span class="co"># Para añadir más información de anotación de los genes:</span>
<span class="no">genesInfo</span> <span class="kw">=</span> <span class="kw pkg">AnnotationDbi</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">select</a></span>(<span class="no">org.Ce.eg.db</span>, <span class="kw">keys</span> <span class="kw">=</span> <span class="no">wormID</span>, <span class="kw">columns</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"ENTREZID"</span>, <span class="st">"SYMBOL"</span>), <span class="kw">keytype</span> <span class="kw">=</span> <span class="st">"WORMBASE"</span>)

<span class="no">genesInfo</span>

<span class="co"># Nos quedamos con la primera coincidencia de WormBase:</span>
<span class="no">posiciones</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="no">genesInfo</span>[,<span class="fl">1</span>]),<span class="no">genesInfo</span>[,<span class="fl">1</span>]) <span class="co"># Para WORMBASE</span>
<span class="no">genesInfo</span> <span class="kw">=</span> <span class="no">genesInfo</span>[<span class="no">posiciones</span>,]

<span class="no">final1</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">WORMBASEID</span> <span class="kw">=</span> <span class="fu"><a href="../reference/getURL_WBC.html">getURL_WBC</a></span>(<span class="no">genesInfo</span>[,<span class="fl">1</span>]), <span class="kw">ENTREZID</span> <span class="kw">=</span><span class="fu"><a href="../reference/getURL_entrez.html">getURL_entrez</a></span>(<span class="no">genesInfo</span>[,<span class="fl">2</span>]), <span class="kw">SYMBOL</span> <span class="kw">=</span> <span class="no">genesInfo</span>[,<span class="fl">3</span>],
                    <span class="kw">P.VALUE</span> <span class="kw">=</span> <span class="no">out1</span>[,<span class="st">'PValue'</span>], <span class="kw">p.BH</span> <span class="kw">=</span> <span class="no">p.values1.BH</span>)
<span class="no">columns</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'WormBaseID'</span>, <span class="st">'EntrezID'</span>, <span class="st">'Symbol'</span>, <span class="st">'p.Value'</span>, <span class="st">'p.BH'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">final1</span>) <span class="kw">=</span> <span class="no">columns</span>

<span class="no">final1</span></pre></body></html></div>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="co"># 2. E.coli - Comamonas:</span>

<span class="no">sel2</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">PRJNA601724</span>)[,<span class="st">"Treatment"</span>] <span class="kw">==</span> <span class="st">"E.coli"</span> <span class="kw">|</span>
  <span class="fu">colData</span>(<span class="no">PRJNA601724</span>)[,<span class="st">"Treatment"</span>] <span class="kw">==</span> <span class="st">"Comamonas"</span>

<span class="no">sel2</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">sel2</span>)

<span class="no">dge2</span> <span class="kw">=</span> <span class="fu">DGEList</span>(<span class="kw">counts</span> <span class="kw">=</span> <span class="fu">assay</span>(<span class="no">PRJNA601724</span>)[,<span class="no">sel2</span>],
            <span class="kw">group</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">PRJNA601724</span>)[<span class="no">sel2</span>,<span class="st">"Treatment"</span>])

<span class="no">dge2c</span> <span class="kw">=</span>  <span class="fu">estimateCommonDisp</span>(<span class="no">dge2</span>) <span class="co"># Asumimos dispersión común para todos los genes</span>
<span class="no">dge2c</span>$<span class="no">common.dispersion</span> <span class="co"># 0.03970863</span>

<span class="no">dge2t</span> <span class="kw">=</span> <span class="fu">estimateTagwiseDisp</span>(<span class="no">dge2c</span>) <span class="co"># Asumimos una dispersión distinta para cada gen.</span>
<span class="no">dge2t</span>$<span class="no">tagwise.dispersion</span> <span class="co"># Para cada gen</span>

<span class="no">dge2cET</span> <span class="kw">=</span> <span class="fu">exactTest</span>(<span class="no">dge2c</span>)
<span class="no">dge2tET</span> <span class="kw">=</span> <span class="fu">exactTest</span>(<span class="no">dge2t</span>)

<span class="fu">topTags</span>(<span class="no">dge2cET</span>)
<span class="fu">topTags</span>(<span class="no">dge2tET</span>) <span class="co"># Nos centraremos en este.</span>

<span class="no">out2</span> <span class="kw">=</span> <span class="fu">topTags</span>(<span class="no">dge2tET</span>, <span class="kw">n</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="fu">assay</span>(<span class="no">se0</span>)))

<span class="no">p.values2</span> <span class="kw">=</span> <span class="no">out2</span>$<span class="no">table</span>[,<span class="st">"PValue"</span>]
<span class="no">p.values2.BH</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span>(<span class="no">p.values2</span>, <span class="st">"BH"</span>)

<span class="no">wormID</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">out2</span>) <span class="co"># Cogemos los ID de los genes y creamos el link a la URL.</span>

<span class="no">final2</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">GENE</span> <span class="kw">=</span> <span class="no">wormID</span>, <span class="kw">P.VALUE</span> <span class="kw">=</span> <span class="no">out2</span>[,<span class="st">'PValue'</span>], <span class="kw">p.BH</span> <span class="kw">=</span> <span class="no">p.values2.BH</span>)

<span class="co"># Para añadir más información de anotación de los genes:</span>
<span class="no">genesInfo</span> <span class="kw">=</span> <span class="kw pkg">AnnotationDbi</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">select</a></span>(<span class="no">org.Ce.eg.db</span>, <span class="kw">keys</span> <span class="kw">=</span> <span class="no">wormID</span>, <span class="kw">columns</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"ENTREZID"</span>, <span class="st">"SYMBOL"</span>), <span class="kw">keytype</span> <span class="kw">=</span> <span class="st">"WORMBASE"</span>)

<span class="no">genesInfo</span>

<span class="co"># Nos quedamos con la primera coincidencia de WormBase:</span>
<span class="no">posiciones</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="no">genesInfo</span>[,<span class="fl">1</span>]),<span class="no">genesInfo</span>[,<span class="fl">1</span>]) <span class="co"># Para WORMBASE</span>
<span class="no">genesInfo</span> <span class="kw">=</span> <span class="no">genesInfo</span>[<span class="no">posiciones</span>,]

<span class="no">final2</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">WORMBASEID</span> <span class="kw">=</span> <span class="fu"><a href="../reference/getURL_WBC.html">getURL_WBC</a></span>(<span class="no">genesInfo</span>[,<span class="fl">1</span>]), <span class="kw">ENTREZID</span> <span class="kw">=</span><span class="fu"><a href="../reference/getURL_entrez.html">getURL_entrez</a></span>(<span class="no">genesInfo</span>[,<span class="fl">2</span>]), <span class="kw">SYMBOL</span> <span class="kw">=</span> <span class="no">genesInfo</span>[,<span class="fl">3</span>],
                    <span class="kw">P.VALUE</span> <span class="kw">=</span> <span class="no">out2</span>[,<span class="st">'PValue'</span>], <span class="kw">p.BH</span> <span class="kw">=</span> <span class="no">p.values2.BH</span>)
<span class="no">columns</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'WormBaseID'</span>, <span class="st">'EntrezID'</span>, <span class="st">'Symbol'</span>, <span class="st">'p.Value'</span>, <span class="st">'p.BH'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">final2</span>) <span class="kw">=</span> <span class="no">columns</span>

<span class="no">final2</span></pre></body></html></div>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="co"># 3. Chryseobacterium - Comamonas:</span>

<span class="no">sel3</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">PRJNA601724</span>)[,<span class="st">"Treatment"</span>] <span class="kw">==</span> <span class="st">"Chryseobacterium"</span> <span class="kw">|</span>
  <span class="fu">colData</span>(<span class="no">PRJNA601724</span>)[,<span class="st">"Treatment"</span>] <span class="kw">==</span> <span class="st">"Comamonas"</span>

<span class="no">sel3</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">sel3</span>)

<span class="no">dge3</span> <span class="kw">=</span> <span class="fu">DGEList</span>(<span class="kw">counts</span> <span class="kw">=</span> <span class="fu">assay</span>(<span class="no">PRJNA601724</span>)[,<span class="no">sel3</span>],
            <span class="kw">group</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">PRJNA601724</span>)[<span class="no">sel3</span>,<span class="st">"Treatment"</span>])

<span class="no">dge3c</span> <span class="kw">=</span>  <span class="fu">estimateCommonDisp</span>(<span class="no">dge3</span>) <span class="co"># Asumimos dispersión común para todos los genes</span>
<span class="no">dge3c</span>$<span class="no">common.dispersion</span> <span class="co"># 0.1519314</span>

<span class="no">dge3t</span> <span class="kw">=</span> <span class="fu">estimateTagwiseDisp</span>(<span class="no">dge3c</span>) <span class="co"># Asumimos una dispersión distinta para cada gen.</span>
<span class="no">dge3t</span>$<span class="no">tagwise.dispersion</span> <span class="co"># Para cada gen</span>

<span class="no">dge3cET</span> <span class="kw">=</span> <span class="fu">exactTest</span>(<span class="no">dge3c</span>)
<span class="no">dge3tET</span> <span class="kw">=</span> <span class="fu">exactTest</span>(<span class="no">dge3t</span>)

<span class="fu">topTags</span>(<span class="no">dge3cET</span>)
<span class="fu">topTags</span>(<span class="no">dge3tET</span>) <span class="co"># Nos centraremos en este.</span>

<span class="no">out3</span> <span class="kw">=</span> <span class="fu">topTags</span>(<span class="no">dge3tET</span>, <span class="kw">n</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="fu">assay</span>(<span class="no">se0</span>)))

<span class="no">p.values3</span> <span class="kw">=</span> <span class="no">out3</span>$<span class="no">table</span>[,<span class="st">"PValue"</span>]
<span class="no">p.values3.BH</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span>(<span class="no">p.values3</span>, <span class="st">"BH"</span>)

<span class="no">wormID</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">out3</span>) <span class="co"># Cogemos los ID de los genes y creamos el link a la URL.</span>

<span class="no">final3</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">GENE</span> <span class="kw">=</span> <span class="no">wormID</span>, <span class="kw">P.VALUE</span> <span class="kw">=</span> <span class="no">out3</span>[,<span class="st">'PValue'</span>], <span class="kw">p.BH</span> <span class="kw">=</span> <span class="no">p.values3.BH</span>)

<span class="co"># Para añadir más información de anotación de los genes:</span>
<span class="no">genesInfo</span> <span class="kw">=</span> <span class="kw pkg">AnnotationDbi</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">select</a></span>(<span class="no">org.Ce.eg.db</span>, <span class="kw">keys</span> <span class="kw">=</span> <span class="no">wormID</span>, <span class="kw">columns</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"ENTREZID"</span>, <span class="st">"SYMBOL"</span>), <span class="kw">keytype</span> <span class="kw">=</span> <span class="st">"WORMBASE"</span>)

<span class="no">genesInfo</span>

<span class="co"># Nos quedamos con la primera coincidencia de WormBase:</span>
<span class="no">posiciones</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="no">genesInfo</span>[,<span class="fl">1</span>]),<span class="no">genesInfo</span>[,<span class="fl">1</span>]) <span class="co"># Para WORMBASE</span>
<span class="no">genesInfo</span> <span class="kw">=</span> <span class="no">genesInfo</span>[<span class="no">posiciones</span>,]

<span class="no">final3</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">WORMBASEID</span> <span class="kw">=</span> <span class="fu"><a href="../reference/getURL_WBC.html">getURL_WBC</a></span>(<span class="no">genesInfo</span>[,<span class="fl">1</span>]), <span class="kw">ENTREZID</span> <span class="kw">=</span><span class="fu"><a href="../reference/getURL_entrez.html">getURL_entrez</a></span>(<span class="no">genesInfo</span>[,<span class="fl">2</span>]), <span class="kw">SYMBOL</span> <span class="kw">=</span> <span class="no">genesInfo</span>[,<span class="fl">3</span>],
                    <span class="kw">P.VALUE</span> <span class="kw">=</span> <span class="no">out3</span>[,<span class="st">'PValue'</span>], <span class="kw">p.BH</span> <span class="kw">=</span> <span class="no">p.values3.BH</span>)
<span class="no">columns</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'WormBaseID'</span>, <span class="st">'EntrezID'</span>, <span class="st">'Symbol'</span>, <span class="st">'p.Value'</span>, <span class="st">'p.BH'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">final3</span>) <span class="kw">=</span> <span class="no">columns</span>

<span class="no">final3</span></pre></body></html></div>
<p>Llegados a este punto ya hemos aplicado el test edgeR. Por lo cual, se han obtenido los genes con expresión diferencial significativa entre las comparaciones dos a dos entre los diferentes grupos. Podemos pasar a crear los informes html utilizando la herramienta ReportingTools.</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="co"># 1. E.coli - Chryseobacterium:</span>
<span class="no">foutput1</span> <span class="kw">=</span> <span class="st">'PRJNA601724_Ecoli_Chryseo.report'</span>
<span class="no">htmlRep1</span> <span class="kw">=</span> <span class="fu">HTMLReport</span>(<span class="kw">shortName</span> <span class="kw">=</span> <span class="no">foutput1</span>, <span class="kw">title</span> <span class="kw">=</span> <span class="no">foutput1</span>, <span class="kw">reportDirectory</span> <span class="kw">=</span> <span class="st">'./reports'</span>)
<span class="fu">publish</span>(<span class="no">final1</span>,<span class="no">htmlRep1</span>)
<span class="fu">finish</span>(<span class="no">htmlRep1</span>)

<span class="co"># 2. E.coli - Comamonas:</span>
<span class="no">foutput2</span> <span class="kw">=</span> <span class="st">'PRJNA601724_Ecoli_Comam.report'</span>
<span class="no">htmlRep2</span> <span class="kw">=</span> <span class="fu">HTMLReport</span>(<span class="kw">shortName</span> <span class="kw">=</span> <span class="no">foutput2</span>, <span class="kw">title</span> <span class="kw">=</span> <span class="no">foutput2</span>, <span class="kw">reportDirectory</span> <span class="kw">=</span> <span class="st">'./reports'</span>)
<span class="fu">publish</span>(<span class="no">final2</span>,<span class="no">htmlRep2</span>)
<span class="fu">finish</span>(<span class="no">htmlRep2</span>)

<span class="co"># 3. Chryseobacterium - Comamonas:</span>
<span class="no">foutput3</span> <span class="kw">=</span> <span class="st">'PRJNA601724_Chryseo_Comam.report'</span>
<span class="no">htmlRep3</span> <span class="kw">=</span> <span class="fu">HTMLReport</span>(<span class="kw">shortName</span> <span class="kw">=</span> <span class="no">foutput3</span>, <span class="kw">title</span> <span class="kw">=</span> <span class="no">foutput3</span>, <span class="kw">reportDirectory</span> <span class="kw">=</span> <span class="st">'./reports'</span>)
<span class="fu">publish</span>(<span class="no">final3</span>,<span class="no">htmlRep3</span>)
<span class="fu">finish</span>(<span class="no">htmlRep3</span>)</pre></body></html></div>
<p>De esta forma se ha conseguido elaborar <strong>tres informes</strong> en los que se muestran los genes que se expresan diferencialmente como resultado de las <em>comparaciones de los grupos dos a dos</em>.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Francisco Martinez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
